src/UnitTests/Analysis/SsaMutatorTests.cs

```diff
@@ -37,111 +39,67 @@ namespace Reko.UnitTests.Analysis
+                stm.Instruction.Accept(codeFormatter);
+            var actual = writer.ToString();
+            if (expected != actual)
+            {
+                Debug.Print(actual);
+                Assert.AreEqual(expected, actual);
             }
         }
 
         [Test]
         public void SsamCallBypass()
         {
-            var sst = BuildSsa(m =>
+            var sp = m.Architecture.StackRegister;
+            var sp_1 = m.Reg("sp_1", sp);
+            var sp_5 = m.Reg("sp_5", sp);
+            var a = m.Reg32("a");
+            var fp = m.Reg32("fp");
+            m.Assign(sp_1, fp);
+            var uses = new Identifier[] { a, sp_1 };
+            var defines = new Identifier[] { sp_5 };
+            var stmCall = m.Call(a, 4, uses, defines);           
+
+            RunSsaMutator(ssam =>
             {
-                var a7 = m.Frame.EnsureRegister(m.Architecture.StackRegister);
-                var a1 = m.Reg32("a1", 0x09);
-                var a3 = m.Reg32("a3", 0x0B);
-                m.Assign(a7, m.Frame.FramePointer);
-                m.Assign(a1, m.Mem32(m.IAdd(a3, 8)));
-                m.Call(a1, 4);
-                m.Return();
+                var call = (CallInstruction)stmCall.Instruction;
+                ssam.AdjustRegisterAfterCall(stmCall, call, sp, 0);
             });
-            var ssam = new SsaMutator(sst.SsaState);
-            var block = sst.SsaState.Procedure.EntryBlock.Succ[0];
-            var stmCall = block.Statements[2];
-            var call = (CallInstruction)stmCall.Instruction;
 
-            ssam.AdjustRegisterAfterCall(stmCall, call, arch.StackRegister, 0);
-            var sExp =
-            #region Expected
-@"fp:fp
-    def:  def fp
-    uses: r63_1 = fp
-r63_1: orig: r63
-    def:  r63_1 = fp
-    uses: call a1_4 (retsize: 4;)	uses: a1_4,a3,r63_1	defs: a1_6,a3_7
-          r63_5 = r63_1
-a3:a3
-    def:  def a3
-    uses: a1_4 = Mem0[a3 + 0x00000008:word32]
-          call a1_4 (retsize: 4;)	uses: a1_4,a3,r63_1	defs: a1_6,a3_7
-Mem0:Global memory
-    def:  def Mem0
-    uses: a1_4 = Mem0[a3 + 0x00000008:word32]
-a1_4: orig: a1
-    def:  a1_4 = Mem0[a3 + 0x00000008:word32]
-    uses: call a1_4 (retsize: 4;)	uses: a1_4,a3,r63_1	defs: a1_6,a3_7
-          call a1_4 (retsize: 4;)	uses: a1_4,a3,r63_1	defs: a1_6,a3_7
-r63_5: orig: r63
-    def:  r63_5 = r63_1
-a1_6: orig: a1
-    def:  call a1_4 (retsize: 4;)	uses: a1_4,a3,r63_1	defs: a1_6,a3_7
-a3_7: orig: a3
-    def:  call a1_4 (retsize: 4;)	uses: a1_4,a3,r63_1	defs: a1_6,a3_7
-// ProcedureBuilder
-// Return size: 0
-void ProcedureBuilder()
-ProcedureBuilder_entry:
-	def fp
-	def a3
-	def Mem0
-	// succ:  l1
-l1:
-	r63_1 = fp
-	a1_4 = Mem0[a3 + 0x00000008:word32]
-	call a1_4 (retsize: 4;)
-		uses: a1_4,a3,r63_1
-		defs: a1_6,a3_7
-	r63_5 = r63_1
-	return
-	// succ:  ProcedureBuilder_exit
-ProcedureBuilder_exit:
+            var sExp = @"
+sp_1 = fp
+call a (retsize: 4;)
+	uses: a,sp_1
+sp_5 = sp_1
```

Comment:
@uxmal Looks better, is not it?

Reply:
I still want to see the ssa state of at least the sp identifier to make sure the ssa state has been updated correctly

Reply:
`CheckUses` method verifiers ssa state. So we'are sure the ssa state has been updated correctly

Reply:
@uxmal I have replaced `CheckUses`  with `Check` which verifies both uses and definitions. So now is no more reason to print `ssa` state

Reply:
Can we change the word `Check` to `Validate` as that is more idiomatic in situations like this where you are validating the "sanity" of the code state. I have no objections to the changes otherwise.

Reply:
OK

Reply:
Done in e9958a30

Reply:
Thanks!